### PR TITLE
Limit clients api to admin-only users

### PR DESCRIPTION
**Pull request by jacqui**
This PR changes who can access the `/pp-api/clients` endpoints in the API, restricting access to admins only. I've updated the tests to reflect this in general, and I've also added an additional test for each endpoint.

### CODE REVIEW

File: squarelet/oidc/viewsets.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ class ClientViewSet(viewsets.ModelViewSet):
 
     queryset = Client.objects.none()
     serializer_class = ClientSerializer
-    permission_classes = (DjangoObjectPermissions,)
+    permission_classes = (IsAdminUser,)
```
Comment (mitchelljkotler) [inline squarelet/oidc/viewsets.py:22]: This should be left as `DjangoObjectPermissions` and the permissions should be controlled through the `squarelet/oidc/rules.py` file

**Comment by TylerFisher** [comment]
My bad, didn't catch that.

File: squarelet/oidc/rules.py [hunk 2]
```diff
@@ -13,7 +13,7 @@ def is_owner(user, client):
     return user == client.owner
 
 
-add_perm("oidc_provider.view_client", is_authenticated & is_owner)
-add_perm("oidc_provider.add_client", is_authenticated)
-add_perm("oidc_provider.change_client", is_authenticated & is_owner)
-add_perm("oidc_provider.delete_client", is_authenticated & is_owner)
+add_perm("oidc_provider.view_client", is_staff)
```
Comment (mitchelljkotler) [inline squarelet/oidc/rules.py:16]: So only staff will create clients, but do you still want users to be able to view a list of their own clients?

Comment (TylerFisher) [reply squarelet/oidc/rules.py:16]: Not at the moment, though I could imagine exposing that at some point.

**Pull request merged**